File: src/main/java/br/com/zumbolovsky/fateapp/domain/DefaultRoles.java
package br.com.zumbolovsky.fateapp.domain;

import br.com.zumbolovsky.fateapp.service.IllegalArgumentException;

import java.util.List;

import static br.com.zumbolovsky.fateapp.config.error.ErrorArguments.ROLE;

public interface DefaultRoles {
    String USER = "USER";
    String STAFF = "STAFF";
    String ADMIN = "ADMIN";
    List<String> ALL_ROLES = List.of(USER, STAFF, ADMIN);
    static String valueOf(String name) {
        return ALL_ROLES.stream()
                .filter(role -> role.equals(name))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException(List.of(name, ROLE)));
    }
}

File: src/main/kotlin/br/com/zumbolovsky/fateapp/domain/DefaultRoles.kt
package br.com.zumbolovsky.fateapp.domain

import br.com.zumbolovsky.fateapp.config.error.ErrorArguments.ROLE
import br.com.zumbolovsky.fateapp.service.IllegalArgumentException

object DefaultRoles {
    const val USER = "USER"
    const val STAFF = "STAFF"
    const val ADMIN = "ADMIN"
    private val ALL_ROLES = listOf(USER, STAFF, ADMIN)
    fun valueOf(name: String) : String =
        ALL_ROLES.find { it == name }
            .takeIf { it != null }
            ?: throw IllegalArgumentException(debugArgs = listOf(name, ROLE))
}